### ISSUE TITLE
Missing icons when opening a my.home-assistant.io link on iPadOS

### ISSUE THREAD

**Issue by Fabian2205** [home-assistant/iOS#1517]
**iOS device model, version and app version**

Model Name: iPad (6. Generation)
Software Version: 14.5 (18E5164h)
App version: 2021.3 (2021.77)

**Home Assistant Core Version**
2021.3.2

**Describe the bug**
When opening a my.home-assistant.io link on my iPad (not on the iPhone) automatically in the Home Assistant app the icons in the sidebar and on the top are hidden (screenshots)

**To Reproduce**
- Open any my.home-assistant.io  link on home-assistant.io
- When the app opens tap on ‚Open link‘
- Then the link opens the page in Home Assistant, but most of the buttons are missing

**Expected behavior**
All buttons are shown 

**Screenshots**

![1E7D12B0-0B08-4DBA-93F9-F5CD0503635D](https://user-images.githubusercontent.com/80009544/110236065-19a1e780-7f34-11eb-82a4-f80f75824fed.png)
![554B3444-5BB4-4D2F-8ABB-0FF85E0D4D4A](https://user-images.githubusercontent.com/80009544/110236066-1b6bab00-7f34-11eb-8680-5d5e2c74c231.png)
![03BDA240-5590-43DB-A3DA-ED3EF0646202](https://user-images.githubusercontent.com/80009544/110236067-1c044180-7f34-11eb-9376-45660501319a.png)
![18C644DE-D9CB-402B-A757-41A8DEFCE081](https://user-images.githubusercontent.com/80009544/110236079-2de5e480-7f34-11eb-90ed-2ccbfe1c2dcb.jpeg)

**Additional context**

**Comment by robbiet480** [comment]
If you open the my link in Safari instead of the app, does the same thing happen?

**Comment by Fabian2205** [comment]
No, it doesn‘t happen in the Safari browser (see screenshot).

![DDA43ACD-3154-41D8-83F3-07EC8FCC6BB3](https://user-images.githubusercontent.com/80009544/110255383-a62fc280-7f93-11eb-9fb1-01b93876d3b1.jpeg)

**Comment by TomBrien** [comment]
Can you try resetting the App's frontend cache (in app go to App Configuration then it's the bottom option there) and try reopening the my.home-assistant link in the app

**Comment by Fabian2205** [comment]
Yes, I tried that out, but nothing changed

**Comment by robiebab** [comment]
I have the same bug. I’m also missings a lot of icons at my dashboard and some in the sidebar.

i removed HA completely also from phone, nevertheless the result is the same after reinstalling

**Comment by westonnovelli** [comment]
Same issue on iPhone. It seems as though its not just icons, but certain texts too. Could be a translations failure or similar? The left menu on mine now says "developer_tools" and "config" where it used to say "Developer Tools" and "Configuration".

**Comment by TomBrien** [comment]
> Same issue on iPhone. It seems as though its not just icons, but certain texts, too. Could be a translations failure or similar? The left menu on mine now says "developer_tools" and "config" where it used to say "Developer Tools" and "Configuration". 
> 
> <img src="https://user-images.githubusercontent.com/8573255/130293501-a4606100-d5ac-4d7a-86cb-ad65b8ee93fe.jpeg" height=500/>

@westonnovelli your issue looks very similar to an issue that was fixed a little while ago (that was indeed to do with loading translations if I remember correctly). Is you Home Assistant core version up to date?

**Comment by westonnovelli** [comment]
> Is you Home Assistant core version up to date?
I believe so: 
```
Core
Version
  core-2021.8.8
Newest Version
  core-2021.8.8
```

**Comment by TomBrien** [comment]
Yep that's up to date? I assume likewise for app? Do you see the same in Safari on iOS?

**Comment by westonnovelli** [comment]
> Yep that's up to date? I assume likewise for app? Do you see the same in Safari on iOS?

Yes, the app is also running latest available. Reinstall of the app did not help. Loading the local or remote HA site in Safari works correctly. 

Over the last 2 days, the app was able to load correctly for a brief usage session (i.e. open app, toggle some lights, check the thermostat, close the app). I don't think I did anything different between correct and incorrect behavior.

**Comment by fmonday** [comment]
I have had the missing icons issue (and more) for many months, usually when switching from a different app back into the companion app on two different iPads. [I filed in the frontend repository ](https://github.com/home-assistant/frontend/issues/9602#issuecomment-901485099)and didn't get any useful feedback.

**Comment by apauly** [comment]
Same thing happens for me as well.

It seems that this is somehow related to SSL.
If I use `http://<LOCAL-IP>:8123` as the internal URL, everything is snappy and looks as expected.
If I use `homeassistant.home` (handled by local DNS server) via http, still everything works as expected.

As soon as I switch to `https`, it is not working anymore on my local network (with an untrusted, self-seigned cert).
Switching to a publicly available URL with a Let's Encrypt cert that connects to my HA instance via an SSH Tunnel doesn't change the behaviour.

In these cases icons and texts are missing as mentioned above, and loading times increase like a lot (feels as if things are reloaded on each page!?).

**Comment by thebatfink** [comment]
I have had this problem too since three years now. It started with my 10” ipad pro, then later with a current gen 12” pro. Parts of the screen do not redraw when it opens, flicking that area of the screen causes it to redraw and display. In 2019 I discussed this in the mobile apps section of discord with zacwest (I think one of the devs), so multiple devices of different generations and several ios versions it has existed. You can search discord for from user thebatfink around 2019.

I also opened a what the heck as its a little more open this year in the hopes of getting some traction, please vote and reply, maybe it will add visibility of this issue. https://community.home-assistant.io/t/wth-why-does-the-ipad-ha-app-ui-sometimes-not-redraw-on-opening/467810/9

**Comment by bgoncal** [comment]
Is this still an issue with latest versions of app and core? Feel free to reopen if needed
